pytket/binders/circuit/Circuit/add_op.cpp

```diff
@@ -517,11 +524,144 @@ void init_circuit_add_op(py::class_<Circuit, std::shared_ptr<Circuit>> &c) {
             return add_box_method(
                 circ, std::make_shared<CircBox>(box), args, kwargs);
           },
-          "Append a :py:class:`CircBox` to the circuit.\n\n:param "
-          "circbox: The box to append\n:param args: The qubits/bits "
-          "to append the box to"
+          "Append a :py:class:`CircBox` to the circuit."
+          "\n\nThe qubits and bits of the :py:class:`CircBox` are wired into "
+          "the circuit in lexicographic order. Bits follow qubits in the order "
+          "of arguments."
+          "\n\n:param circbox: The box to append"
+          "\n:param args: The qubits/bits to append the box to"
           "\n:return: the new :py:class:`Circuit`",
           py::arg("circbox"), py::arg("args"))
+      .def(
+          "add_circbox_regwise",
+          [](Circuit *circ, const CircBox &box, const py_qreg_vector_t &qregs,
+             const py_creg_vector_t &cregs, const py::kwargs &kwargs) {
+            std::vector<UnitID> args;
+
+            for (const QubitRegister &qreg : qregs) {
+              const std::string &name = qreg.name();
+              for (std::size_t i = 0; i < qreg.size(); i++) {
+                args.push_back(Qubit(name, i));
+              }
+            }
+            for (const BitRegister &creg : cregs) {
+              const std::string &name = creg.name();
+              for (std::size_t i = 0; i < creg.size(); i++) {
+                args.push_back(Bit(name, i));
+              }
+            }
+            return add_box_method(
+                circ, std::make_shared<CircBox>(box), args, kwargs);
+          },
+          "Append a :py:class:`CircBox` to the circuit, wiring whole registers "
+          "together."
+          "\n\n:param circbox: The box to append"
+          "\n:param qregs: Sequence of :py:class:`QubitRegister` from the "
+          "outer :py:class:`Circuit`, in order of corresponding registers in "
+          "the :py:class:`CircBox`"
```

Comment:
Should we specify _lexicographic order_ for clarity?

Reply:
Done.